src/structs/2captcha.ts

```diff
@@ -366,6 +366,46 @@ a     *
         }
     }
 
+    /**
+     * Solves a CloudFlare (turnstile) captcha
+     * @param publicKey The turnstile Site Key
+     * @param pageurl The URL to the website the captcha is seen on
+     * @param serviceURL The FunCaptcha Service URL (recommended)
+     * @param extra Extra properties to pass to 2captcha
+     * 
+     * @returns {Promise<CaptchaAnswer>} The result from the solve
+     * @throws APIError
+     * turnstile("0x4AAAAAAADIkEVyoXXXXXXX", "http://mysite.com/page/with/turnstile/")
+     * .then((res) => {
+     *   console.log(res)
+     * })
+     */
+    public async turnstile(siteKey: string, pageURL: string, extra: UserHCaptchaExtra = { }): Promise<CaptchaAnswer> {
```

Comment:
Using `UserHCaptchaExtra` since the optional params are exactly the same as for hCap.

https://2captcha.com/2captcha-api#turnstile